pkg/cloudprovider/providers/aws/aws_loadbalancer.go

```diff
@@ -308,3 +310,82 @@ func (s *AWSCloud) ensureLoadBalancerInstances(loadBalancerName string, lbInstan
 
 	return nil
 }
+
+func getLoadBalancerHostedZone(service *api.Service) string {
+	if zone, ok := service.Annotations[ServiceAnnotationLoadBalancerCnameZone]; ok {
+		return zone + "."
```

Comment:
I think this is going to fall foul of the "be liberal in what we accept" principle.  What if zone already had a dot on the end?  Also, it isn't clear why we choose to add a dot here - I suspect we should do it before talking to Route53 (which is why we want the dot, I believe)